Source/Core/Core/IOS/ES/Identity.cpp

```diff
@@ -69,20 +36,15 @@ IPCCommandResult ES::Encrypt(u32 uid, const IOCtlVRequest& request)
     return GetDefaultReply(ES_EINVAL);
 
   u32 keyIndex = Memory::Read_U32(request.in_vectors[0].address);
-  u8* IV = Memory::GetPointer(request.in_vectors[1].address);
   u8* source = Memory::GetPointer(request.in_vectors[2].address);
   u32 size = request.in_vectors[2].size;
-  u8* newIV = Memory::GetPointer(request.io_vectors[0].address);
+  u8* iv = Memory::GetPointer(request.io_vectors[0].address);
   u8* destination = Memory::GetPointer(request.io_vectors[1].address);
 
-  mbedtls_aes_context AES_ctx;
-  mbedtls_aes_setkey_enc(&AES_ctx, s_key_table[keyIndex], 128);
-  memcpy(newIV, IV, 16);
```

Comment:
Well, here's the thing: IOS doesn't use the "IV" vector at all :P

I too found this weird, but I checked two different IOSes and the ioctlv handler just passes IO vector 1 as the IV parameter. This is true for both the decrypt and encrypt functions. There is absolutely no memcpy in IOS; I don't know where people got that from. And even libogc just passes the same IV pointer for both vectors ("IV" and "new IV").

On the other hand, we're missing a bunch of checks that are *actually* present in IOS (see the TODO comment). 

So yes, this changes behaviour, but it's (1) fixing a wild inaccuracy (2) unlikely to make any difference anyway.